srtcore/core.cpp

```diff
@@ -5498,9 +5498,8 @@ int CUDT::receiveBuffer(char *data, int len)
                 const steady_clock::time_point exptime = steady_clock::now() + milliseconds_from(m_iRcvTimeOut);
                 while (stillConnected() && !m_pRcvBuffer->isRcvDataReady())
                 {
-                    rcond.wait_for(milliseconds_from(m_iRcvTimeOut));
-                    if (steady_clock::now() >= exptime)
-                        break;
+                    if (!rcond.wait_until(exptime)) // NOT means "not received a signal"
+                        break; // timeout
```

Comment:
The implementation of `wait_until()` actually uses `SyncEvent::wait_for()`. This means that spurious wake up can happen, and the condition `if (steady_clock::now() >= exptime)` is not evaluated no more due to the `break` statement.

If the condition `m_pRcvBuffer->isRcvDataReady()` is still false, then connection loss will be detected. Earlier than it should.

Reply:
> The implementation of `wait_until()` actually uses `SyncEvent::wait_for()`. This means that spurious wake up can happen, and the condition `if (steady_clock::now() >= exptime)` is not evaluated no more due to the `break` statement.

As long as this `break` statement is executed - which takes place only in case when the `exptime` time has been achieved. See the condition in `if` here.